File: frontend/src/microservices/blog/objects/BlogCommentPlannerRequest.ts
// BlogCommentPlannerRequest：博客域博客评论请求对象。

export type BlogCommentPlannerRequest = {
  postId: string
  userId: string
  content: string
  parentCommentId?: string | null
  replyToUserId?: string | null
}

export const blogCommentPlannerRequestFromJson = (json: string): BlogCommentPlannerRequest =>
  JSON.parse(json) as BlogCommentPlannerRequest

export const blogCommentPlannerRequestToJson = (value: BlogCommentPlannerRequest): string =>
  JSON.stringify(value)

File: backend/src/main/scala/microservices/blog/objects/BlogCommentPlannerRequest.scala
// BlogCommentPlannerRequest：博客域博客评论请求对象。

package com.typesafe.travel.blog.domain

import io.circe.{Decoder, Encoder}
import io.circe.generic.semiauto.{deriveDecoder, deriveEncoder}

final case class BlogCommentPlannerRequest(
    postId: String,
    userId: String,
    content: String,
    parentCommentId: Option[String],
    replyToUserId: Option[String]
)
object BlogCommentPlannerRequest:
  given sourceEncoder: Encoder[BlogCommentPlannerRequest] = deriveEncoder
  given sourceDecoder: Decoder[BlogCommentPlannerRequest] = deriveDecoder